pkg/pool-manager/mackey.go

```diff
@@ -0,0 +1,22 @@
+package pool_manager
+
+import "net"
+
+type macKey struct {
+	normalizedMacAddress string
+}
+
+// NewMacKey creates a macKey struct containing a mac string
+// that can be used for different mac string format comparison.
+// It uses net.ParseMAC function to parse the given macAddress
+// and then stores its String() value.
+// The given macAddress MUST have already been validated.
+func NewMacKey(macAddress string) macKey {
+	// we can ignore the error here since the string value has already been validated
+	hwMacAddress, _ := net.ParseMAC(macAddress)
+	return macKey{normalizedMacAddress: hwMacAddress.String()}
+}
+
+func (m *macKey) String() string {
```

Comment:
Since we are not changing the value of the object, there is no need to use a pointer to the object
```suggestion
func (m macKey) String() string {
```

This has an added value of being able to call the `String()` right after creating the object: `NewMacKey(MacAddress).String()`